frontend/src/scenes/insights/ActionFilter/ActionFilterRow.js

```diff
@@ -280,40 +282,46 @@ function MathPropertySelector(props) {
         ({ value }) => value[0] !== '$' && value !== 'distinct_id' && value !== 'token'
     )
 
-    const overlay = () => {
-        return (
-            <Menu onClick={({ item }) => props.onMathPropertySelect(props.index, item.props['data-value'])}>
-                {applicableProperties.map(({ value, label }) => {
-                    return (
-                        <Menu.Item
-                            key={`math-property-${value}-${props.index}`}
-                            data-attr={`math-property-${value}-${props.index}`}
-                            data-value={value}
-                        >
-                            <Tooltip
-                                title={
-                                    <>
-                                        Calculate {MATHS[props.math].name.toLowerCase()} from property{' '}
-                                        <code>{label}</code>. Note that only {props.name} occurences where{' '}
-                                        <code>{label}</code> is set and a number will be taken into account.
-                                    </>
-                                }
-                                placement="right"
-                            >
-                                {label}
-                            </Tooltip>
-                        </Menu.Item>
-                    )
-                })}
-            </Menu>
-        )
-    }
-
     return (
-        <Dropdown overlay={overlay}>
-            <Button data-attr={`math-property-selector-${props.index}`} style={{ marginTop: 8 }}>
-                {props.mathProperty || 'Select property'} <DownOutlined />
-            </Button>
-        </Dropdown>
+        <SelectGradientOverflow
+            showSearch
+            style={{ width: 150 }}
+            onChange={(_, payload) => props.onMathPropertySelect(props.index, payload && payload.value)}
+            className="property-select"
+            value={props.mathProperty}
+            onSearch={(input) => {
+                setInput(input)
+                if (!optionsCache[input] && !isOperatorFlag(operator)) {
+                    loadPropertyValues(input)
+                }
+            }}
+            data-attr="prop-val"
+            dropdownMatchSelectWidth={350}
+            placeholder={'Select property'}
+        >
+            {applicableProperties.map(({ value, label }) => {
```

Comment:
Currently the properties shown in order of first seen - let's sort them alphabetically!